tests/psoc6/pin.py.exp

```diff
@@ -0,0 +1,21 @@
+Pin constructor invoked
+Pin class object instantiated
+init helper function called
+Direction: 1, Drive:6, Value:1
+machine pin print
+Pin:111 or P13_7, Mode=OUT, Pull=None, Value=1
+machine pin call
+machine pin call
+None
+None
```

Comment:
All this traces are available by default? 
I would consider some of them only relevant for a debugging mode.

Reply:
The first two lines should only be available in debug mode. The switch to enable debug mode is active in the `ports/psoc6` branch and also this branch, which should not be the case. See here: https://github.com/jaenrig-ifx/micropython/blob/e76692e849b54fa5c08cdeddadc7b1641c6b03c9/ports/psoc6/mpconfigport.h#L134

The last two lines (None's) are a return of MPY functions which is defined in their guidelines so those should normally be there irrespective of debug mode active or not.

We should disable the switch `#define MICROPY_LOGGER_DEBUG                    (0)` and have the exp files updated for the tests.

Reply:
Yes I agree. I was testing out what happens in debug mode and did not revert. My bad. Done now! Thank you for highlighting.

Reply:
Understood.
I mean, for test, if you see convenient to have the test in debug mode, we can keep it like that... But by the productive version it should be disabled. 
If that is the case we could pass the flag during compilation instead and use ifndef guards.

Reply:
mmm yes, but we would have 2 .exp files. That would require to modify the run-tests.py utility.
The only problem with the logger mode enable, is that that is not really the actual program, as all the debugger traces have an impact on performance: memory, timing, code size... 
So, maybe we are adding more complexity than the value we get by running the tests in debug mode?

Reply:
I agree with @jaenrig-ifx. We should not touch the `run_tests.py` anymore.